paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodService.java

```diff
@@ -0,0 +1,233 @@
+
+					for (String value : paymentMethodGroups) {
+						filteredMethodsList.addAll(Arrays.stream(methods)
+								.filter(method -> method.getGroup().equalsIgnoreCase(value))
+								.collect(Collectors.toList()));
+					}
+				}
+			}
+		}
+
+		if (!filteredMethodsList.isEmpty()) {
+			methods = filteredMethodsList.toArray(new PaymentMethodDto[0]);
+		}
+
+		return methods;
+	}
+
+	private List<PaymentMethodFilter> getFiltersEnabledForNamespace(String namespace) {
+		List<PaymentMethodFilter> filtersForNamespace = new ArrayList<>();
+
+		for (PaymentMethodFilter methodFilter : PaymentMethodFilter.getAll()) {
+			String orderTypeFilterEnabledList = env.getRequiredProperty("enabled_namespaces.payment_method_filter." + methodFilter);
+
+			if (orderTypeFilterEnabledList.contains(namespace)) {
+				filtersForNamespace.add(methodFilter);
+			}
+		}
+
+		return filtersForNamespace;
+	}
+
+	public String getFilterKeys(PaymentMethodFilter filter, GetPaymentMethodListRequest request) {
+		OrderDto orderDto = request.getOrderDto();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			return orderDto.getType();
+		}
+
+		throw new CommonApiException(
+				HttpStatus.INTERNAL_SERVER_ERROR,
+				new Error("unknown-payment-method-filter-type", "unknown payment method filter type")
+		);
+	}
+
+	// namespace can be used as a parameter if different namespaces need to return different values for filter type
+	public HashMap<String, List<String>> getFilterValuesMap(PaymentMethodFilter filter) {
+		HashMap<String, List<String>> values = new HashMap<>();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			values.put(OrderType.SUBSCRIPTION, Collections.singletonList(PaymentType.CREDIT_CARDS));
+		}
+
+		return values;
+	}
+
+	public List<PaymentMethodDto> getAllPaymentMethods() {
+		Iterable<fi.hel.verkkokauppa.payment.model.PaymentMethod> paymentMethods = paymentMethodRepository.findAll();
+
+		List<PaymentMethodDto> paymentMethodDtos = StreamSupport.stream(paymentMethods.spliterator(), false)
+				.map(paymentMethod -> mapper.convertValue(paymentMethod, PaymentMethodDto.class))
+				.collect(Collectors.toList());
+		return paymentMethodDtos;
+	}
+
+	public PaymentMethodDto getPaymenMethodByCode(String code) {
+		fi.hel.verkkokauppa.payment.model.PaymentMethod exsistingMethod = paymentMethodRepository.findByCode(code).stream()
+				.filter(paymentMethod -> paymentMethod.getCode().equals(code))
+				.findFirst()
+				.orElseThrow(() -> new CommonApiException(
+						HttpStatus.NOT_FOUND,
+						new Error("payment-method-not-found", "payment method with code [" + code + "] not found")
+				));
+		PaymentMethodDto dto = mapper.convertValue(exsistingMethod, PaymentMethodDto.class);
+		return dto;
+	}
+
+	public PaymentMethodDto createNewPaymentMethod(PaymentMethodDto dto) {
+		paymentMethodRepository.findByCode(dto.getCode()).stream()
+				.filter(paymentMethod -> paymentMethod.getCode().equals(dto.getCode()))
```

Comment:
We dont need this kind of checks in core, these are done in experience layer.(ExpApi) 

If this is left you should move this to new function named `isDuplicatePaymentMethod`